python/ray/rllib/dqn/dqn_evaluator.py

```diff
@@ -59,14 +59,28 @@ def update_target(self):
 
     def sample(self):
         obs, actions, rewards, new_obs, dones = [], [], [], [], []
-        for _ in range(self.config["sample_batch_size"]):
+        for _ in range(
+                self.config["sample_batch_size"] + self.config["n_step"] - 1):
             ob, act, rew, ob1, done = self._step(self.global_timestep)
             obs.append(ob)
             actions.append(act)
             rewards.append(rew)
             new_obs.append(ob1)
             dones.append(done)
-        return SampleBatch({
```

Comment:
isn't a `return` needed?

Reply:
Yeah, that's what I get for pasting code from another PR. I should actually re-run this experiment to test if n_step improves efficiency as expected.

Reply:
n_step=9 seems to consistently do better on cartpole (and n_step=3, 6, also better than n_step=1).